sklearn/ensemble/gradient_boosting.py

```diff
@@ -634,7 +646,8 @@ def _update_terminal_region(self, tree, terminal_regions, leaf, X, y,
         numerator = np.sum(y_ * sample_weight * np.exp(-y_ * pred))
         denominator = np.sum(sample_weight * np.exp(-y_ * pred))
 
-        if denominator == 0.0:
+        # prevents overflow and division by zero
+        if abs(denominator) < 1e-150:
```

Comment:
Instead of `1e-150` you could use `np.finfo(np.float32).eps`... There is precedent for it [here](https://github.com/scikit-learn/scikit-learn/blob/8570622a4450fe1ee3c683601454a7189dcccc14/sklearn/linear_model/theil_sen.py#L69)